src/chapter03-getting-started.md

```diff
@@ -110,27 +110,75 @@ Note that the echo request in step 6 is just a simple usage example. Once the co
-    3. Allocate an NVM context structure and initialize it with the configuration from step 2.2 using `wh_Nvm_Init()`
-3. Allocate and initialize a crypto context structure for the server
-4. Initialize wolfCrypt (before initializing the server)
-5. Allocate and initialize a server config structure and bind the comm server configuration, NVM context, and crypto context to it
-6. Allocate a server context structure and initialize it with the server configuration using `wh_Server_Init()`
-7. Set the server connection state to connected using `wh_Server_SetConnected()` when the underlying transport is ready to be used for client communication (see [TODO](todo) for more information)
-8. Process client requests using `wh_Server_HandleRequestMessage()`
+2.  Allocate and initialize a crypto context structure for the server
+3. Initialize wolfCrypt (before initializing the server)
+4. Allocate and initialize a server config structure and bind the comm server configuration, NVM context, and crypto context to it
+5. Allocate a server context structure and initialize it with the server configuration using `wh_Server_Init()`
+6. Set the server connection state to connected using `wh_Server_SetConnected()` when the underlying transport is ready to be used for client communication (see [TODO](todo) for more information)
+7. Process client requests using `wh_Server_HandleRequestMessage()`
+
+The server may be configured to support NVM object storage using NVM flash configuration. Include the steps to [initialize NVM](./chapter04-functional-components.md#NVM-Architecture) on the server after step 1.
 
 ```c
  /* TODO: basic server configuration example */
+#include <string.h> /* for memcmp() */
+#include "wolfhsm/server.h"  /* Server API (includes comm config) */
+#include "wolfhsm/wh_transport_mem.h" /* transport implementation */
+
+/* Step 1.1: Allocate and initialize the shared memory transport configuration */
+/* Shared memory transport configuration */
+static whTransportMemConfig transportMemCfg = { /* shared memory config */ };
+
+/* Shared memory transport context (state) */
+whTransportMemServerContext transportMemServerCtx = {0};
+
+/* Callback structure that binds the abstract comm transport interface to
+ * our concrete implementation */
+whTransportServerCb transportMemServerCb = {WH_TRANSPORT_MEM_SERVER_CB};
+
+/* Step 1.2: Allocate a comm server configuration structure and bind to the
+ * transport */
+/* Configure the server comms to use the selected transport configuration*/
+whCommServerConfig commServerCfg[1] = {{
+        .transport_cb   = transportMemServerCb,
+        .transport_context  = (void*)transportMemServerCtx,
+        .transport_config   = (void*)transportMemCfg,
+        .server_id          = 456, /* unique client identifier */
+}};
+
+//* Step 2: Allocate and initialize a crypto context structure*/
+crypto_context crypto[1]{{
+    .devID = 123;
```

Comment:
Can you instead show the devId initialized to `INVALID_DEVID`, and then add a comment saying something like: "can also be set to your devId for a custom crypto callback"

check out [the crypto callback doc section](https://www.wolfssl.com/documentation/manuals/wolfssl/chapter06.html#crypto-callbacks-cryptocb) for background